File: LeetCode/Study Plan/Practice Problems/Easy/Remove Duplicates From Sorted List/Go/remove_dup_from_sorted_list.go
package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func New(val int) *ListNode { return &ListNode{Val: val, Next: nil} }

func remove_dup(head *ListNode) *ListNode {
	temp := head
	prev := head
	for temp != nil {
		if temp.Val == prev.Val {
			for temp != nil && prev.Val == temp.Val {
				temp = temp.Next
			}
			prev.Next = temp
			prev = prev.Next
		} else {
			prev = temp
			temp = temp.Next

		}
	}
	return head
}

func main() {
	head := New(1)
	head.Next = New(1)
	head.Next.Next = New(2)
	fmt.Println(remove_dup(head))
}

File: LeetCode/Easy/Remove Duplicates From Sorted List/Java/removeDupFromSortedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode temp = head;
        while (temp != null) {
            ListNode curr = temp;
            while(curr != null && temp.val == curr.val) {
                curr = curr.next;
            }
            temp.next = curr;
            temp = temp.next;
        }
        return head;
    }
}
